coderd/userauth.go

```diff
@@ -437,6 +437,41 @@ func (api *API) postChangePasswordWithOneTimePasscode(rw http.ResponseWriter, r
 	}
 }
 
+// ValidateUserPassword validates the complexity of a user password and that it is secured enough.
+//
+// @Summary Validate user password
+// @ID validate-user-password
+// @Security CoderSessionToken
+// @Produce json
+// @Accept json
+// @Tags Authorization
+// @Param request body codersdk.ValidateUserPasswordRequest true "Validate user password request"
+// @Success 200 {object} codersdk.ValidateUserPasswordResponse
+// @Router /users/validate-password [post]
+func (*API) validateUserPassword(rw http.ResponseWriter, r *http.Request) {
```

Comment:
nit about referring to all of this stuff as "user password" validation. the password is not _yet_ attached to a user, so it's not a user password, it's just a password, so I'd expect it to be called `validatePassword` and so on.